core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -90,6 +90,8 @@ private static Map<String, String> persistedProperties(Map<String, String> rawPr
     persistedProperties.put(
         TableProperties.PARQUET_COMPRESSION,
         TableProperties.PARQUET_COMPRESSION_DEFAULT_SINCE_1_4_0);
+    persistedProperties.put(
```

Comment:
By adding this, we need to update a few tests (like Hive), as we see now `"write.delete.granularity"="file"` in the metadata persistence.

I propose to update the tests.

Maybe worth to add a test to check how to read "previous" files which don't contain the property ?

Reply:
Yes, we'll definitely need to update the failing Hive tests in their current form since it looks like they currently have expectations on the persisted properties. 

>Maybe worth to add a test to check how to read "previous" files which don't contain the property ?

In terms of general metadata file reading though, the core library will just read the metadata file with or without the property (since the properties is just a string-string map). In terms of how engine integrations handle the missing property, at the moment only Spark really leverages this property with a default option in case it's not defined, and we actually already have a [test](https://github.com/apache/iceberg/blob/main/spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/TestSparkWriteConf.java#L140C15-L140C21) for verifying this default handling in case it's missing. 

I'll double check the other integrations though

Reply:
can we add a simple test to `TestTableMetadata` to make sure this is properly set on new tables and not set on existing ones?